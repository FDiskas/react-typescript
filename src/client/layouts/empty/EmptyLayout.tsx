import * as React from 'react';
import * as styles from '../default/DefaultLayout.scss';

export interface IEmptyLayoutProps {}

export interface IEmptyLayoutState {}

export class EmptyLayout extends React.Component<IEmptyLayoutProps, IEmptyLayoutState> {
  constructor(props: IEmptyLayoutProps) {
    super(props);
  }

  render() {
    return (
      <div className={styles.layout}>
        Component name: EmptyLayout
        {this.props.children}
      </div>
    );
  }
}
