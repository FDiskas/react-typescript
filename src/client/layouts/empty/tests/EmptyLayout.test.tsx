import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { EmptyLayout } from '../EmptyLayout';

describe('EmptyLayout.js', () => {
    it('renders without crashing', () => {
        shallow(<EmptyLayout/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<EmptyLayout/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
