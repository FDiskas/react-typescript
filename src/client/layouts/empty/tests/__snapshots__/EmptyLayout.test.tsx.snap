// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyLayout.js matches snapshot 1`] = `
<div
  className="layout"
>
  Component name: EmptyLayout
</div>
`;
